Mage/src/main/java/mage/abilities/effects/common/continuous/CastFromHandWithoutPayingManaCostEffect.java

```diff
@@ -28,14 +28,18 @@ public CastFromHandWithoutPayingManaCostEffect() {
     }
 
     public CastFromHandWithoutPayingManaCostEffect(FilterCard filter, boolean fromHand) {
-        super(Duration.WhileOnBattlefield, Outcome.Detriment);
-        this.filter = filter;
-        this.fromHand = fromHand;
+        this(filter, fromHand, Duration.WhileOnBattlefield);
         staticText = "You may cast " + filter.getMessage()
                 + (fromHand ? " from your hand" : "")
                 + " without paying their mana costs";
     }
 
+    public CastFromHandWithoutPayingManaCostEffect(FilterCard filter, boolean fromHand, Duration duration) {
+        super(duration, Outcome.Detriment);
+        this.filter = filter;
+        this.fromHand = fromHand;
```

Comment:
Missing `staticText` code. Use one main constructor with code setup and max params and other additional constructors with `this(...)` call.

Reply:
fixed